src/aiida/transports/transport.py

```diff
@@ -103,6 +114,20 @@ def __init__(self, *args, **kwargs):
         # for accessing the identity of the underlying machine
         self.hostname = kwargs.get('machine')
 
+    @abc.abstractmethod
+    def open(self):
+        """Opens a transport channel
+
+        :raises InvalidOperation: if the transport is already open.
+        """
+
+    @abc.abstractmethod
+    def close(self):
```

Comment:
If this class is not a metaclass (which provide as the interface), then make no sense to use `@abc.abstractmethod`.